args.go

```diff
@@ -345,31 +387,41 @@ func delAllArgs(args []argsKV, key string) []argsKV {
 	return args
 }
 
-func setArgBytes(h []argsKV, key, value []byte) []argsKV {
-	return setArg(h, b2s(key), b2s(value))
+func setArgBytes(h []argsKV, key, value []byte, noValue bool) []argsKV {
+	return setArg(h, b2s(key), b2s(value), noValue)
 }
 
-func setArg(h []argsKV, key, value string) []argsKV {
+func setArg(h []argsKV, key, value string, noValue bool) []argsKV {
 	n := len(h)
 	for i := 0; i < n; i++ {
 		kv := &h[i]
 		if key == string(kv.key) {
-			kv.value = append(kv.value[:0], value...)
+			if ArgsNoValue == noValue {
+				kv.value = kv.value[:0]
+			} else {
+				kv.value = append(kv.value[:0], value...)
+			}
+			kv.noValue = noValue
 			return h
 		}
 	}
-	return appendArg(h, key, value)
+	return appendArg(h, key, value, noValue)
 }
 
-func appendArgBytes(h []argsKV, key, value []byte) []argsKV {
-	return appendArg(h, b2s(key), b2s(value))
+func appendArgBytes(h []argsKV, key, value []byte, noValue bool) []argsKV {
+	return appendArg(h, b2s(key), b2s(value), noValue)
 }
 
-func appendArg(args []argsKV, key, value string) []argsKV {
+func appendArg(args []argsKV, key, value string, noValue bool) []argsKV {
 	var kv *argsKV
 	args, kv = allocArg(args)
 	kv.key = append(kv.key[:0], key...)
-	kv.value = append(kv.value[:0], value...)
+	if ArgsNoValue == noValue {
```

Comment:
I would prefer `noValue == ArgsNoValue` which reads easier and is more in line with the rest of the codebase.